src/fft/domain.rs

```diff
@@ -406,7 +406,7 @@ mod tests {
 
     #[test]
     fn dusk_bytes_evaluation_domain_serde() {
-        let eval_domain = EvaluationDomain::new(1 << 13 - 1)
+        let eval_domain = EvaluationDomain::new((1 << 13) - 1)
```

Comment:
Your change improves the code, but seems it should be `1 << 13`. We expect here a power of 2 and this `- 1` simply doesn't make any sense.

Reply:
We have two issues here then.
1. We should ensure a failure if we do not give a power of two: but the tests are passing.
2. Before it was basically `1 << 12` and the tests were still passing. So either we need `1 << 12` and not `1 << 13`, or our code and tests are wrong.

Reply:
For the time being I'm going to use directly `1 << 12` as discussed, however we would need to improve the API that takes numbers that needs to be power of two / next power of two.